RustBridge/RustBridge.cpp

```diff
@@ -149,11 +149,20 @@ namespace WPEFramework {
 			// The expectation is that the JavaScript service opens up a connection to us, so we can forward the 
 			// incomming requests, to be handled by the Service.
 			if (_javascriptService == 0) {
+				/*
 				Web::ProtocolsArray protocols = channel.Protocols();
 				if (std::find(protocols.begin(), protocols.end(), string(_T("json"))) != protocols.end()) {
 					_javascriptService = channel.Id();
 					assigned = true;
 				}
+				*/
```

Comment:
This is dropping functionality that is, according to the websocket spec allowed, see: https://www.rfc-editor.org/rfc/rfc6455#section-1.9 . In R2, this header was treated as a string (so only 1 subprotocol could be considered, in R3, on request from some customers, it was required to follow the spec more closely and make this, as the spec suggests a list of identifiers, hence why compilation in R2 fails as that is not a list, it is a string (see https://github.com/rdkcentral/Thunder/blob/R2/Source/websocket/WebSocketLink.h#L649).
Suggest to , in R2, to stll check the subprotocol header but just "scan" for "json" so that it is mandataory at least for the service to use this header as an upgrade scenario, alwasy prepare for the future, don't code for it but think about it ;-)
So a good replacement in R2 could be (pseudeo code):
'''
if (channel.Protocol().find("json") != string::npos) {
    _javascriptService = channel.Id();
    assigned = true;
}
'''

Reply:
agreed, replaced with a simple search for "json" in the subprotocol string as you recommended